service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -402,7 +401,7 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
                         attributes.addAll(deserializedAttributes);
                         // This enforces the limit on the count of attributes returned from the traversal, instead of
                         // number of vertices traversed. To do the latter will require traversing the graph twice.
-                        checkTraversalLimitOrFail(attributes);
+                        checkTraversalLimitOrFail(entity, attributes);
```

Comment:
Disregard my earlier comment. I think after playing with the code I understand the problem a bit better. However, there are still things to consider. First of all, realize that because the code is turned to a stream, no actual work is done by the code until you call `forEach()` which will begin to drive the actual traversal. Once you realize that, you will also realize that the limit() is not really entirely needed. You can just keep track of the number of times your `forEach()` lambda is called and simply throw an exception when the limit is reached.